chronos/handles.nim

```diff
@@ -105,10 +105,6 @@ proc createAsyncSocket*(domain: Domain, sockType: SockType,
   if not setSocketBlocking(handle, false):
     close(handle)
     return asyncInvalidSocket
-  when defined(macosx) and not defined(nimdoc):
-    if not setSockOpt(AsyncFD(handle), SOL_SOCKET, SO_NOSIGPIPE, 1):
-      close(handle)
-      return asyncInvalidSocket
```

Comment:
Hmm. Not sure these are correct.

To avoid `SIGPIPE` during `send` on a socket that is being closed during the send, one of these needs to be done:
- `SO_NOSIGPIPE` on the socket itself (only on BSD)
- `MSG_NOSIGNAL` for each `send` call (Linux)
- `signal(SIGPIPE, SIG_IGN)` on the process (POSIX, but blocks it for the entire process not just for sockets)

https://github.com/tarsnap/libcperciva even goes so far as to temporarily ignore SIGPIPE / restore the handler on each syscall, but that is likely too crazy here xD

https://github.com/nim-lang/Nim/blob/v1.6.0/lib/pure/asyncdispatch.nim#L1655-L1664 uses the same approach as was used here before (with exceptions instead of `invalidSocket` though – here, the caller is throwing the exception though, so same end result).

Reply:
Maybe less disruptive:

Check for the error after the `setSockOpt`, throw `TransportAbortedError` on `EINVAL` and `TransportOsError` otherwise? For `setSocketBlocking`, only `TransportOsError` is good enough (that's what the caller currently makes of of both of the errors).

Reply:
If we unable to perform `setSockOpt` on socket it means we can't continue (we could not make socket non-blocking). So we definitely should close this socket and return an error, that connection could not be made for whatever reason. `EINVAL` has here 2 meanings, so i wish to return `EINVAL` instead of describing what we see here.